vending-app/src/main/kotlin/com/google/android/finsky/splitinstallservice/SplitInstallManager.kt

```diff
@@ -0,0 +1,337 @@
+            Log.w(TAG, "No Google account found")
+            throw RuntimeException("No Google account found")
+        } else for (account: Account in accounts) {
+            oauthToken = try {
+                getAuthToken(AccountManager.get(context), account, AUTH_TOKEN_SCOPE).getString(AccountManager.KEY_AUTHTOKEN)
+            } catch (e: AuthenticatorException) {
+                Log.w(TAG, "Could not fetch auth token for account $account")
+                null
+            }
+            if (oauthToken != null) {
+                break
+            }
+        }
+        return oauthToken ?: throw RuntimeException("oauthToken is null")
+    }
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    private fun checkSplitInstalled(callingPackage: String, splitName: String): Boolean {
+        if (!splitInstallRecord.containsKey(callingPackage)) return false
+        return splitInstallRecord[callingPackage]?.find { it.first == splitName }?.third != STATUS_UNKNOWN
+    }
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    private fun updateSplitInstallRecord(callingPackage: String, triple: Triple<String, String, Int>) {
+        splitInstallRecord[callingPackage]?.let { triples ->
+            val find = triples.find { it.first == triple.first }
+            find?.let { triples.remove(it) }
+            triples.add(triple)
+        } ?: run {
+            val triples = ArraySet<Triple<String, String, Int>>()
+            triples.add(triple)
+            splitInstallRecord[callingPackage] = triples
+        }
+    }
+
+    private fun notify(context: Context) {
+        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            notificationManager.createNotificationChannel(
+                NotificationChannel(NOTIFY_CHANNEL_ID, NOTIFY_CHANNEL_NAME, NotificationManager.IMPORTANCE_DEFAULT)
+            )
+        }
+        NotificationCompat.Builder(context, NOTIFY_CHANNEL_ID).setSmallIcon(android.R.drawable.stat_sys_download)
+            .setContentTitle(context.getString(R.string.split_install, context.getString(R.string.app_name))).setPriority(NotificationCompat.PRIORITY_DEFAULT)
+            .setDefaults(
+                NotificationCompat.DEFAULT_ALL
+            ).build().also {
+                notificationManager.notify(SPLIT_INSTALL_NOTIFY_ID, it)
+            }
+    }
+
+    private fun Context.splitSaveFile() = File(filesDir, FILE_SAVE_PATH)
+
+    private fun sendCompleteBroad(context: Context, packageName: String, intent: Intent) {
+        Log.d(TAG, "sendCompleteBroadcast: intent:$intent")
+        val extra = Bundle().apply {
+            putInt(KEY_STATUS, 5)
+            putInt(KEY_ERROR_CODE, 0)
+            putInt(KEY_SESSION_ID, 0)
+            putLong(KEY_TOTAL_BYTES_TO_DOWNLOAD, intent.getLongExtra(KEY_BYTES_DOWNLOADED, 0))
+            putString(KEY_LANGUAGES, intent.getStringExtra(KEY_LANGUAGE))
```

Comment:
It doesn't seem like this intent would have this field, given that it is not set upon creation of the intent in `installSplitPackage`.